sample/CustomHandlerRetry/server.js

```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+
+const app = express();
+const port = process.env.FUNCTIONS_HTTPWORKER_PORT; //|| 5001;
+
+app.use(bodyParser.json()) // for parsing application/json
+
+app.get('/api/httptrigger', (req, res) => {
+    let retryCount = req?.body?.Metadata?.RetryContext?.RetryCount || 0;
+    let maxRetry = req?.body?.Metadata?.RetryContext?.MaxRetryCount;
```

Comment:
I tried this out and it looks like because this is the server, if an error is thrown the request will come back as a 500.

Reply:
Regarding the discussion above about throwing to trigger a retry I think the request coming back with a 500 is expected behaviour.

```js
let retryCount = req?.body?.Metadata?.RetryContext?.RetryCount || 0;
let maxRetryCount = req?.body?.Metadata?.RetryContext?.MaxRetryCount;
let exceptionMessage = req?.body?.Metadata?.RetryContext?.Exception?.message || '';

if (maxRetryCount != maxRetries || (retryCount > 0 && !exceptionMessage.includes(errorString))) {
    res.status(500).send();
} else {
    console.log('JavaScript HTTP trigger function processed a request. retryCount: ' + retryCount);

    if (retryCount < maxRetries) {
        throw new Error(errorString);
    }

    let json = JSON.stringify({ functionName: req.url.replace("/", ""), retryCount, maxRetryCount});
    res.send(json);
}
```

Looking at the code snippet above, my expectation on how we would test this would be:

1. Set the max retry value to something like 3
2. Trigger the function the first time, it should fail, return 500, the retry count should increment
3. Trigger the function the second time, it should fail, return 500, the retry count should increment
4. Trigger the function the third time, it should fail, return 500, the retry count should increment
5. And when we trigger the forth time, and now the retry count should be == to our max retires, at this point we should pass the function and return 200 OK

My only confusion here is:

1. What is the difference between retryCount and maxRetryCount? What are they supposed to be used for? Why do we have two counts for a retry? Or maybe the naming is off and it doesn't mean what I think it means
2. Who is actually handling the retry? The caller of the function or the function itself. If it's the function, that doesn't make much sense to me. If it's the caller, then our test should be managing the retry count and incrementing it so that we can test the use case outlined above

Reply:
retryCount - current retry count
maxRetries - max retry count specified by customer

retryCount  - incrementing retryCount is in webjobs sdk:
https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutorExtensions.cs#L64

Reply:
The behavior @madelinegordon is currently observing is related to fact she was using HTTP forwarding. I've followed up with her on this.